appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/repositories/DiscoveryAppsRemote.kt

```diff
@@ -28,30 +29,28 @@ class Network {
         httpClient = httpClientBuilder.build()
     }
 
-    private fun getRequest(url: String): Request = Request.Builder().url(url).build()
-
-    fun getDiscoveryApps() {
-        httpClient.newCall(getRequest(GET_DISCOVERY_APPS_STAGE_URL)).enqueue(object : Callback {
+    fun getDiscoveryAppsServerData(callback: OperationResultCallback<EcosystemAppResponse>) {
+        httpClient.newCall(getRequest(GET_DISCOVERY_APPS_PROD_URL)).enqueue(object : Callback {
             override fun onFailure(call: Call, e: IOException) {
-                //TODO
-                Log.d("", "response failed")
+                callback.onError("server failed get discoveryApps ${e.message}")
```

Comment:
Pass here the actual exception, and the repo should create a ModuleException with some logic.